dubbo-metrics/dubbo-metrics-default/src/main/java/org/apache/dubbo/metrics/collector/sample/ThreadRejectMetricsCountSampler.java

```diff
@@ -47,28 +48,23 @@ public ThreadRejectMetricsCountSampler(DefaultMetricsCollector collector) {
         this.collector.addSampler(this);
     }
 
-    public void addMetricName(String name){
+    public void addMetricName(String name) {
         this.metricNames.add(name);
-        this.initMetricsCounter(name,name);
+        this.initMetricsCounter(name, name);
         samplesChanged.set(true);
     }
 
     @Override
     public List<MetricSample> sample() {
-        List<MetricSample> metricSamples = new ArrayList<>();
-        metricNames.stream().forEach(name->collect(metricSamples,name));
-        return metricSamples;
-    }
-
-
-    private void collect(List<MetricSample> list, String metricName) {
-        count(list, metricName, MetricsKey.THREAD_POOL_THREAD_REJECT_COUNT);
-    }
-
-    private <T extends Metric> void count(List<MetricSample> list, String metricName, MetricsKey metricsKey) {
-        getCount(metricName).filter(e -> !e.isEmpty())
-            .ifPresent(map -> map.forEach((k, v) ->
-                list.add(getGaugeMetricSample(metricsKey, k, THREAD_POOL, v, AtomicLong::get))));
+        return metricNames
+            .stream()
+            .map(metricName -> getCount(metricName)
+                .entrySet()
+                .stream()
+                .map(ele -> getGaugeMetricSample(MetricsKey.THREAD_POOL_THREAD_REJECT_COUNT, ele.getKey(), THREAD_POOL, ele.getValue(), AtomicLong::get))
+                .collect(Collectors.toList()))
+            .filter(CollectionUtils::isNotEmpty)
+            .flatMap(Collection::stream).collect(Collectors.toList());
```

Comment:
This nesting is too complicated. Please define the code in the map as a method and give it a meaningful name.

Reply:
fixed